consumer-java/src/main/java/org/apache/pulsar/client/impl/WeightedConsumerBuilder.java

```diff
@@ -103,12 +104,21 @@ public WeightedConsumerBuilder<T> topic(String topic, Integer weight) {
     }
 
     public WeightedConsumerBuilder<T> topics(Map<String, Integer> topicWeights) {
-        Preconditions.checkArgument(topicWeights != null && topicWeights.size() > 0, "non-empty topic-weight map required");
+        checkArgument(topicWeights != null && topicWeights.size() > 0, "non-empty topic-weight map required");
         for (Map.Entry<String, Integer> entry : topicWeights.entrySet()) {
             topic(entry.getKey(), entry.getValue());
         }
         return this;
     }
+    private String getSubscriptionName(ConsumerConfigurationData<T> conf){
```

Comment:
Actually, you should encode the sub name here... or replace `/` with another string constant, like `___` or something..
Just returning second part of the sub can cause clashes in case of 2 subs like `A/sub` and `B/sub` consuming from same topic